src/main/java/com/aws/greengrass/certificatemanager/CertificateManager.java

```diff
@@ -115,9 +115,13 @@ public String getCaPassPhrase() {
      * @throws KeyStoreException if unable to access KeyStore
      * @throws CsrProcessingException if unable to process CSR
      */
+    // the service certificate subscribe function is invoked on the thread starting broker, i.e. nucleus
+    // lifecycle management thread, the callback function is invoked in CertificateGenerator class function
+    // which could happen on the thread scheduled by connectivity info monitor
     @SuppressWarnings("PMD.AvoidCatchingGenericException")
     public void subscribeToServerCertificateUpdates(@NonNull String csr, @NonNull Consumer<X509Certificate> cb)
             throws KeyStoreException, CsrProcessingException {
+        logger.atInfo().log("Subscribe to service certificate update");
```

Comment:
I don't think this adds value. I'd prefer more information on the caller. Once we expose this via IPC we'll have that additional info. For now, I think we log this in moquette (or should, if we don't)

Reply:
Yeah, caller side should log the call, but server side should log the trace as well IMO. Inside the function call, it has more context, `CSR` should be loggable in this situation to provide more info.